packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -15,159 +16,348 @@ import {
-    this.buttonClickEvent.emit(event);
+  /**
+   * Implements keyboard interface from Keyboard Support and guidance from Role, Property, State, and Tabindex Attribute at
+   * https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html
+   *
+   * Enter key was modified to also fire a submit event.
+   */
+  private handleListboxKeyDown = (event: KeyboardEvent, index: number) => {
+    const options = this.el.shadowRoot.querySelectorAll(
+      '[role="option"]',
+    ) as NodeListOf<HTMLLIElement>;
+
+    switch (event.key) {
+      case 'ArrowUp':
+        const lastOption = options[options.length - 1];
+        if (index === 0) {
+          if (!lastOption) return;
+          this.selectSuggestion(lastOption);
+        } else {
+          if (!options[index - 1]) return;
+          this.selectSuggestion(options[index - 1]);
+        }
+        break;
+      case 'ArrowDown':
+        const firstOption = options[0];
+        if (index === options.length - 1) {
+          if (!firstOption) return;
+          this.selectSuggestion(firstOption);
+        } else {
+          if (!options[index + 1]) return;
+          this.selectSuggestion(options[index + 1]);
+        }
+        break;
+      case 'Enter':
+        this.inputRef.value = options[index].innerText;
+        this.inputRef.focus();
+        this.inputRef.removeAttribute('aria-activedescendant');
+        this.isListboxOpen = false;
+        this.handleSubmit();
+        break;
+      case 'Escape':
+        this.inputRef.value = '';
+        this.inputRef.focus();
+        this.inputRef.removeAttribute('aria-activedescendant');
+        this.isListboxOpen = false;
+        break;
+      case 'Home':
+        this.inputRef.focus();
+        this.inputRef.setSelectionRange(0, 0);
+        break;
+      case 'ArrowRight':
+      case 'End':
+        this.inputRef.focus();
+        this.inputRef.setSelectionRange(
+          this.inputRef.value.length + 1,
+          this.inputRef.value.length + 1,
+        );
+        break;
+      case 'ArrowLeft':
+        this.inputRef.focus();
+        this.inputRef.setSelectionRange(
+          this.inputRef.value.length,
+          this.inputRef.value.length,
+        );
+        break;
+      default:
+        this.inputRef.focus();
+        break;
+    }
   };
 
-  private handleButtonFocusEvent = (event: FocusEvent) => {
-    this.buttonFocusEvent.emit(event);
+  /**
+   * Formats suggested characters to bold
+   */
+  private formatSuggestion = (suggestion: string) => {
+    const lowercaseSuggestion = suggestion.toLowerCase();
+    const inputValue = this.inputRef.value.toLowerCase();
+    if (lowercaseSuggestion.includes(inputValue)) {
+      return (
+        <Fragment>
+          {inputValue}
+          <strong>{lowercaseSuggestion.replace(inputValue, '')}</strong>
+        </Fragment>
+      );
+    }
+    return <strong>{suggestion}</strong>;
```

Comment:
I want to double check my understanding. We're formatting the suggestions based on the `suggestions` change instead of the input change, but consumers of the component _should_ change  `suggestions` based on the input change, is that right? I ask because I expected the bolding to continue creeping into the tail end of "benefits" as I deleted characters and it didn't happen:

![bolded suggestions after "benefit"](https://user-images.githubusercontent.com/2008881/164276439-dc5812d6-cc39-43dd-a710-8b7b2208c889.png)

I think that's fine, but we should make sure we document when we apply the formatting so that teams understand what they should do.

Reply:
This has been updated so that the suggestion formatting is updated on user input to provide the "creeping" you described.